parser/src/python.lalrpop

```diff
@@ -472,197 +431,168 @@ OrPattern: ast::Pattern = {
     <location:@L> <pattern:ClosedPattern> <patterns:("|" <ClosedPattern>)+> <end_location:@R> => {
         let mut patterns = patterns;
         patterns.insert(0, pattern);
-        ast::Pattern::new(
-            location..
-            end_location,
-            ast::PatternMatchOr { patterns }
+        ast::Pattern::MatchOr(
+            ast::PatternMatchOr { patterns, range: (location..end_location).into() }
         )
     }
 }
 
 ClosedPattern: ast::Pattern = {
-    <location:@L> <node:LiteralPattern> <end_location:@R> => ast::Pattern::new(
-        location..
-        end_location,
-        node,
-    ),
-    <location:@L> <node:CapturePattern> <end_location:@R> => ast::Pattern::new(
-        location..
-        end_location,
-        node,
-    ),
-    <location:@L> <node:StarPattern> <end_location:@R> => ast::Pattern::new(
-        location..
-        end_location,
-        node,
-    ),
-    <location:@L> <node:ValuePattern> <end_location:@R> => ast::Pattern::new(
-        location..
-        end_location,
-        node,
-    ),
-    <location:@L> <node:SequencePattern> <end_location:@R> => ast::Pattern::new(
-        location..
-        end_location,
-        node,
-    ),
-    <location:@L> <node:MappingPattern> <end_location:@R> => ast::Pattern::new(
-        location..
-        end_location,
-        node,
-    ),
-    <location:@L> <node:ClassPattern> <end_location:@R> => ast::Pattern::new(
-        location..
-        end_location,
-        node,
-    ),
+    <node:LiteralPattern> => node,
```

Comment:
Can we remove this grammar entirely?

Reply:
cc @charliermarsh

Reply:
Do we not need it? It seems like it's necessary here to define the "enum" of valid sub-types.

Reply:
I don't know. I only know how to pipe location information through lalrpop

Reply:
I think this is a sort of match in lalrpop. Not only making location information.